File: Comp2150Assignment1/Node.java
// CLASS: Node
//
// Author: Adrian Hizon, 7867173
//
// PURPOSE: These are the nodes that are used
//          in the linked list.
//-----------------------------------------

public class Node 
{
	private Node next;
	private NodeData data;
	
	public Node(NodeData data)
	{
		next = null;
		this.data = data;
	}
	
	public Node getNext()
	{
		return next;
	}
	
	public void setNext(Node newNode)
	{
		next = newNode;
	}
	
	public NodeData getData()
	{
		return data;
	}
	
	public void printNode()
	{
		data.printData();
	}
	
//	public abstract void printNode();
//	public abstract Node deepCopy();
}

File: Comp2150Assignment2/Node.cpp
// CLASS: Node
//
// Author: Adrian Hizon, 7867173
//
// REMARKS: A node to be used in a data structure
//-----------------------------------------

#include "Node.h"
#include "NodeData.h"
class DataNode;

Node::Node(NodeData * newData)
{
	this->data = newData;
}

void Node::setNext(Node * nextNode)
{
	next = nextNode;
}

NodeData * Node::getData()
{
	return data;
}

Node * Node::getNext()
{
	return next;
}

void Node::printNode()
{
	data->printData();
}